main/lsp/requests/go_to_implementation.cc

```diff
@@ -0,0 +1,152 @@
+#include "main/lsp/requests/go_to_implementation.h"
+#include "core/lsp/QueryResponse.h"
+#include "main/lsp/json_types.h"
+#include "main/lsp/lsp.h"
+#include <cstddef>
+#include <memory>
+
+using namespace std;
+
+namespace sorbet::realmain::lsp {
+
+namespace {
+struct MethodImplementationResults {
+    vector<core::Loc> locations;
+    unique_ptr<ResponseError> error;
+};
+} // namespace
+
+unique_ptr<ResponseError> makeInvalidParamsError(std::string error) {
+    return make_unique<ResponseError>((int)LSPErrorCodes::InvalidParams, error);
+}
+
+const MethodImplementationResults findMethodImplementations(const core::GlobalState &gs, core::SymbolRef method) {
+    if (!method.data(gs)->isMethod() || !method.data(gs)->isAbstract()) {
+        return {.error = {makeInvalidParamsError(
+                    "Go to implementation can be used only for methods or references of abstract classes")}};
+    }
+
+    vector<core::Loc> locations;
+    auto owner = method.data(gs)->owner;
+    if (!owner.isClassOrModule())
+        return {.error = {makeInvalidParamsError("Abstract method can only be inside a class or module")}};
+
+    auto owningClassSymbolRef = owner.asClassOrModuleRef();
+    auto childClasses = owningClassSymbolRef.getSubclasses(gs, false);
+    auto methodName = method.data(gs)->name;
+    for (const auto &childClass : childClasses) {
+        auto methodImplementation = childClass.data(gs)->findMember(gs, methodName);
+        locations.push_back(methodImplementation.data(gs)->loc());
+    }
+
+    return {.locations = {locations}};
```

Comment:
If you go the route of adding the default-initialized `res` at the top of the method, you could make the following change as well:

```suggestion
    res.locations = locations;
    return res;
```